### PR TITLE
Fix/indexer metadata

### PR DESCRIPTION
**Pull request by lamcc21**
- refactor where and how indexer entity is created 
- update subql/contract-sdk version

### CODE REVIEW

File: src/mappings/utils.ts [hunk 1]
```diff
@@ -113,43 +114,55 @@ export async function upsertEraValue(
...
-  } else {
-    indexer.totalStake = await upsertEraValue(
-      eraManager,
-      indexer.totalStake,
-      amount,
-      operation,
-      applyInstantly
-    );
-  }
+  if (!indexer) return;
```
Comment (heyacherry) [inline src/mappings/utils.ts]:
```
 if (!indexer) {
    indexer = Indexer.create({
      id: indexerAddress,
      totalStake: await upsertEraValue(
      .....
```
or throw assert error

Comment (lamcc21) [reply src/mappings/utils.ts]: Is this needed?, @HZHAIX and I discussed that Indexer only needs to be created in handleCommissionRate and handleRegisterIndexer.

Comment (lamcc21) [reply src/mappings/utils.ts]: handleCommissionRate should be the only case where it comes before the handleRegisterIndexer event.

Comment (mzxyz) [reply src/mappings/utils.ts]: Aha, `addDelegationEvent` also trigger align with them. Just discuss with Cherry, for this part we will report exception and include `assert` action in `reportException`

File: src/mappings/staking.ts [hunk 2]
```diff
@@ -267,27 +272,16 @@ export async function handleSetCommissionRate(
...
-      },
-      commission: {
-        era: -1,
-        value: BigInt(0).toJSONType(),
-        valueAfter: BigInt(0).toJSONType(),
-      },
-      active: true,
+    indexer = await createIndexer({
+      address,
+      active: false,
```
Comment (mzxyz) [inline src/mappings/staking.ts]:
```suggestion
      active: true,
```

**Comment by mzxyz** [comment]
@lachlan1m we need add assert at the bottom of `reportException`:

```
export async function reportException(
  handler: string,
  eventLogIdx: number,
  eventBlock: number,
  error: string
): Promise<void> {
  const id = `${handler}:${eventBlock}:${eventLogIdx}`;

  const exception = Exception.create({
    id,
    error: error || `Error: ${id}`,
    handler,
    createBlock: eventBlock,
  });

  await exception.save();

 assert(false, id);
}
```

**Comment by mzxyz** [comment]
Also remove `await createWithdrawl` in `handleWithdrawCancelled` and `handleWithdrawClaimed`

File: src/mappings/staking.ts (hunk 2 above)
Comment (lamcc21) [reply src/mappings/staking.ts]: typo updated

**Comment by mzxyz** [comment]
Awesome work, last thing need to update,  as you already use the latest contract sdk, need to update the contract addresses in project.yml as well

File: src/mappings/utils.ts [hunk 3]
```diff
@@ -203,4 +226,5 @@ export async function reportException(
   });
 
   await exception.save();
+  assert(false, id);
```
Comment (heyacherry) [inline src/mappings/utils.ts]:
```suggestion
  assert(false, `${id}:Error: ${error});` );
```

**Pull request merged**